### DISCUSSION TITLE
Memory limit for MAGMA on Ubuntu

### DISCUSSION THREAD

**Discussion by OverwatchGirl** [magma/magma#11527] [Q&A]
Hello everyone,

I tried to install MAGMA on Windows 10 but it was no success, so i decided to install it on an Ubuntu virtual machine, but i started wondering whether the amount memory that i attributed to my virtual machine will be enough to run MAGMA. 

Could you, guys, enlighten me about the  ideal memory space necessary to run MAGMA ?

Thank you so much.

**Comment by sebathomas** [comment]
Hello!

Which parts of Magma are you trying to run? Is it the dev setup of AGW and Orc8r as described in [the guide](https://magma.github.io/magma/docs/next/basics/quick_start_guide) or something else? The memory requirements depend on which components will be running.

In any case, I would recommend trying it in a native Ubuntu instead of a virtual machine. It's likely you will run into performance issues otherwise.

**Reply by Neudrino** [reply]
You can also have a look at the deployment guide for the different components. E.g. the [deployment guide for the Access Gateway](https://magma.github.io/magma/docs/lte/deploy_install). You will find deployment information of the other components next to it.

As pointed out by @sebathomas, the exact requirements depend on multiple factors (especially also load). Using the resources detailed in the quick start guide to start with might be a good idea though. For cloud deployments I have seen people using Amazon t3a instances with 4 cores. They might have over-provisioned though.
